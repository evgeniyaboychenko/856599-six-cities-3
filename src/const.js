export const CardType = {
  NEAR: `near-places`,
  CITY: `cities`
};
