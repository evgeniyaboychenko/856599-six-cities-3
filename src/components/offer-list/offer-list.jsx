import React from 'react';
import PropTypes from 'prop-types';
import {OfferCard} from '../offer-card/offer-card.jsx';
import {SortType} from '../../const.js';

const sortOffers = (offers, sortType) => {
  switch (sortType) {
    case (SortType.DEFAULT):
      return offers;
    case (SortType.LOW_TO_HIGH):
      return offers.sort((a, b) => (a.price - b.price));
    case (SortType.HIGH_TO_LOW):
      return offers.sort((a, b) => (b.price - a.price));
    case (SortType.TOP_RATED_FIRST):
      return offers.sort((a, b) => (b.rating - a.rating));
  }
  return offers;
};

const OfferList = ({offers, cardType, activeSortItem}) => {
  let offerList = offers.slice();
  return <>{sortOffers(offerList.slice(), activeSortItem).map((offerCard) => {
    const {id} = offerCard;
    return <OfferCard
      offerCard = {offerCard}
      key = {id}
      cardType = {cardType}
    />;
  })} </>;
};

OfferList.propTypes = {
  offers: PropTypes.array.isRequired,
  cardType: PropTypes.string.isRequired,
  activeSortItem: PropTypes.string.isRequired
};

export default React.memo(OfferList);
