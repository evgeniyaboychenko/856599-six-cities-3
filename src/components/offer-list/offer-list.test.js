import React from 'react';
import renderer from 'react-test-renderer';
import {OfferList} from './offer-list.jsx';
import {CardType} from '../../const.js';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const OFFER_IMAGES = [`room.jpg`, `apartment-01.jpg`];
const OFFER_NAMES = [`Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFER_TYPES = [`apartment`, `privet room`];

const offerCards = [
  {
    id: `1`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[0],
    rating: 2,
    price: 100,
    type: OFFER_TYPES[0],
    isPremium: true,
  },
  {
    id: `2`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    type: OFFER_TYPES[1],
    isPremium: false,
  }
];

const offersNear = [
  {
    id: `11`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[0],
    rating: 2,
    price: 100,
    type: OFFER_TYPES[0],
    isPremium: true,
  },
  {
    id: `22`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    type: OFFER_TYPES[1],
    isPremium: false,
  },
  {
    id: `23`,
    photos: OFFER_IMAGES,
    name: OFFER_NAMES[1],
    rating: 3,
    price: 120,
    type: OFFER_TYPES[1],
    isPremium: false,
  }
];


describe(`OfferList component render correctly`, () => {
  it(`should OfferList render correctly for offer cities`, () => {
    const store = mockStore({
      offers: offerCards,
      activeSortItem: `Popular`,
      offersNear: []
    });

    const tree = renderer.create(
        <Provider store={store}>
          <OfferList
            offerCards = {offerCards}
            onHeaderCardClick = {() => {}}
            cardType = {CardType.CITY}
            activeSortItem = {`Popular`}
            offersNear = {offersNear}
          />
        </Provider>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should OfferList render correctly for offer near`, () => {
    const store = mockStore({
      offers: offerCards,
      activeSortItem: `Popular`,
      offersNear
    });
    const tree = renderer.create(
        <Provider store={store}>
          <OfferList
            offerCards = {offerCards}
            onHeaderCardClick = {() => {}}
            cardType = {CardType.NEAR}
            activeSortItem = {`Popular`}
            offersNear = {offersNear}
          />
        </Provider>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
