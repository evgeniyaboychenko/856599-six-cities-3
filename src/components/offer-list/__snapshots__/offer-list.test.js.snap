// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList component render correctly should OfferList render correctly for offer cities 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 40,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseOver={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 60,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        privet room
      </p>
    </div>
  </article>,
]
`;

exports[`OfferList component render correctly should OfferList render correctly for offer near 1`] = `
Array [
  <article
    className="near-places__place-card place-card"
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 40,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>,
  <article
    className="near-places__place-card place-card"
    onMouseOver={[Function]}
  >
    
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              href="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 60,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        privet room
      </p>
    </div>
  </article>,
]
`;
